p2p/pkg/rpc/validator/service.go

```diff
@@ -0,0 +1,192 @@
+	apiURL          string
+	validatorRouter ValidatorRouterContract
+	logger          *slog.Logger
+	metrics         *metrics
+}
+
+func NewService(apiURL string, validatorRouter ValidatorRouterContract, logger *slog.Logger) *Service {
+	return &Service{
+		apiURL:          apiURL,
+		validatorRouter: validatorRouter,
+		logger:          logger,
+		metrics:         newMetrics(),
+	}
+}
+
+type FinalityCheckpointsResponse struct {
+	Data struct {
+		PreviousJustified struct {
+			Epoch string `json:"epoch"`
+		} `json:"previous_justified"`
+		CurrentJustified struct {
+			Epoch string `json:"epoch"`
+		} `json:"current_justified"`
+		Finalized struct {
+			Epoch string `json:"epoch"`
+		} `json:"finalized"`
+	} `json:"data"`
+}
+
+type ProposerDutiesResponse struct {
+	Data []struct {
+		Pubkey string `json:"pubkey"`
+		Slot   string `json:"slot"`
+	} `json:"data"`
+}
+
+func (s *Service) GetEpoch(
+	ctx context.Context,
+	_ *validatorapiv1.EmptyMessage,
+) (*validatorapiv1.EpochResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, s.apiURL+"/eth/v1/beacon/states/head/finality_checkpoints", nil)
+	if err != nil {
+		s.logger.Error("creating request", "error", err)
+		return nil, status.Errorf(codes.Internal, "creating request: %v", err)
+	}
+
+	req.Header.Set("accept", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		s.logger.Error("making request", "error", err)
+		return nil, status.Errorf(codes.Internal, "making request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		s.logger.Error("unexpected status code", "status", resp.StatusCode)
+		return nil, status.Errorf(codes.Internal, "unexpected status code: %v", resp.StatusCode)
+	}
+
+	var checkpointsResp FinalityCheckpointsResponse
+	if err := json.NewDecoder(resp.Body).Decode(&checkpointsResp); err != nil {
+		s.logger.Error("decoding response", "error", err)
+		return nil, status.Errorf(codes.Internal, "decoding response: %v", err)
+	}
+
+	currentEpoch, err := strconv.ParseUint(checkpointsResp.Data.CurrentJustified.Epoch, 10, 64)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "parsing current justified epoch: %v", err)
+	}
+
+	previousJustifiedEpoch, err := strconv.ParseUint(checkpointsResp.Data.PreviousJustified.Epoch, 10, 64)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "parsing previous justified epoch: %v", err)
+	}
+
+	finalizedEpoch, err := strconv.ParseUint(checkpointsResp.Data.Finalized.Epoch, 10, 64)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "parsing finalized epoch: %v", err)
+	}
+
+	s.logger.Info("fetched epoch data", "current_epoch", currentEpoch, "current_justified_epoch", currentEpoch, "previous_justified_epoch", previousJustifiedEpoch, "finalized_epoch", finalizedEpoch)
+	s.metrics.FetchedEpochDataCount.Inc()
+
+	return &validatorapiv1.EpochResponse{
+		CurrentEpoch:           currentEpoch + 1,
```

Comment:
Do we need to expose all three of these values to the user? Is there some tradeoffs we should communicate to them?